### PR TITLE
[Sources] reset --hard rather than git pull

### PR DESCRIPTION
**Pull request by DanToml**
Getting sources ready for squashing of the specs repo

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! 
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* WIP: [Sources] reset --hard rather than git pull 
  [DanToml](https://github.com/DanToml)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by orta** [comment]
👍

**Comment by DanToml** [comment]
@orta updated to explicitly fetch origin in case of weird local configs. should be good to 🚢 now.

**Comment by orta** [comment]
Yeah, this looks legit - @segiddins think this conceptually makes sense?

File: lib/cocoapods/sources_manager.rb [hunk 1]
```diff
@@ -104,7 +104,8 @@ def git(args, include_error: false)
 
     def update_git_repo(show_output = false)
       Config.instance.with_changes(:verbose => show_output) do
-        git!(['-C', repo, 'pull', '--ff-only'])
+        git!(%W(-C #{repo} fetch origin))
+        git!(%W(-C #{repo} reset --hard origin/master))
```
Comment (segiddins) [inline lib/cocoapods/sources_manager.rb]: May want to grab the branch and use that instead of master?

Comment (DanToml) [inline lib/cocoapods/sources_manager.rb]: Yeah that makes sense

**Comment by dnkoutso** [comment]
LGTM.

**Pull request merged**